_scripts/ProcessLocalesPlugin.js

```diff
@@ -37,66 +38,88 @@ class ProcessLocalesPlugin {
 
       compilation.hooks.additionalAssets.tapPromise('process-locales-plugin', async (_assets) => {
 
-        // While running in the webpack dev server, this hook gets called for every incrememental build.
+        // While running in the webpack dev server, this hook gets called for every incremental build.
         // For incremental builds we can return the already processed versions, which saves time
         // and makes webpack treat them as cached
-        if (IS_DEV_SERVER && this.cache.length > 0) {
-          for (const { filename, source } of this.cache) {
-            compilation.emitAsset(filename, source, { minimized: true })
-          }
-        } else {
-          const promises = []
-
-          for (const { locale, data } of this.locales) {
-            promises.push(new Promise(async (resolve) => {
-              if (Object.prototype.hasOwnProperty.call(data, 'Locale Name')) {
-                delete data['Locale Name']
+        const promises = []
+        this.loadLocales(true)
+
+        Object.values(this.locales).forEach((localeEntry) => {
+          const { locale, data, mtimeMs } = localeEntry
+          promises.push(new Promise(async (resolve) => {
+            if (IS_DEV_SERVER) {
+              const cacheEntry = this.cache[locale]
+              if (cacheEntry != null) {
+                const { filename, source, mtimeMs: cachedMtimeMs } = cacheEntry
+                if (cachedMtimeMs === mtimeMs) {
+                  compilation.emitAsset(filename, source, { minimized: true })
+                  resolve()
+                  return
+                }
               }
+            }
```

Comment:
Please add some whitespace to make this easier to read.

```suggestion
        Object.values(this.locales).forEach((localeEntry) => {
          const { locale, data, mtimeMs } = localeEntry

          promises.push(new Promise(async (resolve) => {
            if (IS_DEV_SERVER) {
              const cacheEntry = this.cache[locale]

              if (cacheEntry != null) {
                const { filename, source, mtimeMs: cachedMtimeMs } = cacheEntry

                if (cachedMtimeMs === mtimeMs) {
                  compilation.emitAsset(filename, source, { minimized: true })
                  resolve()
                  return
                }
              }
            }
```